### ISSUE TITLE
Does not support POCO serialisation

### ISSUE THREAD

**Issue by fmannhardt** [amzn/ion-dotnet#121]
The readme indicates that I could serialise any POCO object using:
```
//Serialize an object to byte array
byte[] ionBytes = IonSerialization.Binary.Serialize(experiment);
```

However, I cannot find the class `IonSerialization`.

**Comment by dhhoang** [comment]
It seems like the serialization features have been excluded from the 1.0 release: https://github.com/amzn/ion-dotnet/releases/tag/v1.0.0

Is there any plan to support this feature in the upcoming release? This is useful for a lot of .NET users.

**Comment by tgregg** [comment]
Yes, we removed this API for the 1.0 release to reduce the API surface area. If there is enough demand for it, it could be added in a future release.

For now, I will update the readme to avoid the confusion.

**Comment by adrianiftode** [comment]
I would like to dump a bunch of objects into a Ledger database and I'm thinking a POCO serializer would help. Is there a timeline for adding this feature back?

**Comment by dhhoang** [comment]
I am also having an use case where this is feature is needed.
@tgregg  what are the steps required to implement & test this one?

**Comment by tgregg** [comment]
@dhhoang We need to resurrect the code from this branch: https://github.com/amzn/ion-dotnet/tree/bench-and-serialization-projects

It can be submitted as a PR with its associated tests. Once we've reviewed and merged it, we can create a new release that includes it.

**Comment by ChiefInnovator** [comment]
I really like this project but without the POCO serialization, it is almost useless for general adoption.

**Comment by tgregg** [comment]
@ChiefInnovator The following library may help! https://github.com/amzn/ion-object-mapper-dotnet/
